llvm/lib/CodeGen/MachineVerifier.cpp

```diff
@@ -1788,7 +1788,7 @@ void MachineVerifier::verifyPreISelGenericInstruction(const MachineInstr *MI) {
     }
 
     uint64_t SrcMinLen = SrcTy.getElementCount().getKnownMinValue();
-    if (SrcTy.isScalable() && DstTy.isScalable() &&
+    if (SrcTy.isScalable() == DstTy.isScalable() &&
```

Comment:
We've gone in a circle. We need a `report` if `SrcTy.isScalable() != DstTy.isScalable()`

Reply:
Why? We haven’t specified the behavior for this case have we? I was thinking to have a follow up pr that synced the mixed case with the lang ref.

Reply:
Well I was thinking you need to either disallow the mixed case or check that we don't have the unsupported mixed case of scalable dest and fixed source. So some check needs to exist.

Reply:
Let's start out by not allowing it? We could probably change our mind down the line if we go in that direction (opposed to allowing it now and disallowing in the future).